### PR TITLE
[#124] Switch to new find module for python

### LINKED ISSUE #124: FindPythonInterp.cmake and FindPythonLibs.cmake deprecated in CMake 3.12.
We should switch to `FindPython.cmake`, because the `Development` component provides a complete `Python::Python` target and does not require the interpreter component to define the version variables.

### PR DESCRIPTION
**Pull request by SwooshyCueb**
Closes #124

### CODE REVIEW

**Review by korydraughn — COMMENTED** [review]
The changes look fine to me.

Has this been tested on each OS?

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.5.0 FATAL_ERROR) #CPACK_DEBIAN_<COMPONENT>_PACKAGE_NAME
+cmake_minimum_required(VERSION 3.12 FATAL_ERROR) #FindPython
```
Comment (trel) [inline CMakeLists.txt:1]: what does this do?

Comment (SwooshyCueb) [reply CMakeLists.txt:1]: The `FindPython` module was introduced in CMake 3.12. I'm just changing the comment to reflect why the CMake version requirement was bumped.

**Comment by SwooshyCueb** [comment]
I've only tested on Ubuntu 18. It might be a bit before I get around to testing on other distros, if someone wants to try and beat me to it

**Comment by SwooshyCueb** [comment]
@alanking has tested this on other distros, so I've \#'d

**Pull request merged**